src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -179,6 +180,16 @@ internal void OnSolutionBuild(object sender, UIContextChangedEventArgs e)
                     return;
                 }
 
+                // explicitly start solution crawler if it didn't start yet.
+                // we have a bug where we don't start solution cralwer until a file is opened (package is loaded)
+                // https://github.com/dotnet/roslyn/issues/35514
+                // until that is fixed, we need to do this for the case where user build solution right after opening solution
+                // without any file opened. "EnsureRegistration" is no-op if same workspace is already registered.
```

Comment:
Maybe a question: if we fix the other bug, what's harmful with this behavior? Why is this code dependent upon the other code for correctness?